ios/BT/ExposureManager.swift

```diff
@@ -239,153 +242,58 @@ final class ExposureManager: NSObject {
-    
     var lastProcessedUrlPath: String = .default
     var processedFileCount: Int = 0
+    var unpackedKeysURLs: [URL] = []
 
-    // Disallow concurrent exposure detection,
-    // because if allowed we might try to detect the same diagnosis keys more than once
-    guard !isDetectingExposures else {
-      finish(
-        .failure(ExposureError.default("Detection Already in Progress")),
-        processedFileCount: processedFileCount,
-        lastProcessedUrlPath: lastProcessedUrlPath,
-        progress: progress,
-        completionHandler: completionHandler
-      )
-      return progress
-    }
-
-    isDetectingExposures = true
-
-    // Reset file capacity to 15 if > 24 hours have elapsed since last reset
-    self.updateRemainingFileCapacity()
-
-    // Abort if daily file capacity is exceeded
-    guard btSecureStorage.userState.remainingDailyFileProcessingCapacity > 0 else {
-      finish(
-        .success([]),
-        processedFileCount: processedFileCount,
-        lastProcessedUrlPath: lastProcessedUrlPath,
-        progress: progress,
-        completionHandler: completionHandler
-      )
-      return progress
-    }
-
-    apiClient.requestString(IndexFileRequest.get,
-                            requestType: .downloadKeys) { result in
-      let dispatchGroup = DispatchGroup()
-
-      switch result {
-      case let .success(indexFileString):
-        let remoteURLs = indexFileString.gaenFilePaths
-        let targetUrls = self.urlPathsToProcess(remoteURLs)
-        lastProcessedUrlPath = targetUrls.last ?? .default
-        processedFileCount = targetUrls.count
-        for remoteURL in targetUrls {
-          dispatchGroup.enter()
-          self.apiClient.downloadRequest(DiagnosisKeyUrlRequest.get(remoteURL), requestType: .downloadKeys) { result in
-            switch result {
-            case .success (let package):
-              self.downloadedPackages.append(package)
-            case .failure(let error):
-              self.finish(.failure(error),
-                          processedFileCount: processedFileCount,
-                          lastProcessedUrlPath: lastProcessedUrlPath,
-                          progress: progress,
-                          completionHandler: completionHandler)
-              return
-            }
-            dispatchGroup.leave()
-          }
-        }
-
-      case let .failure(error):
-        self.finish(.failure(error),
-                    processedFileCount: processedFileCount,
-                    lastProcessedUrlPath: lastProcessedUrlPath,
-                    progress: progress,
-                    completionHandler: completionHandler)
-        return
+    Promise<[Exposure]>(on: .global()) { () -> [Exposure] in
+      if self.isDetectingExposures {
+        // Disallow concurrent exposure detection,
+        // because if allowed we might try to detect the same diagnosis keys more than once
+        throw ExposureError.default("Detection Already in Progress")
```

Comment:
general question, will these errors be user-facing? Ideally this would fail silently and the redundant detection cycle would be simply be aborted 🤔

Reply:
on my understanding we are only showing error notifications in debug mode
check postExposureDetectionErrorNotification